script.js

```diff
@@ -1,4 +1,456 @@
+    concatenateNum = Number(concatenateString);
+
+    return concatenateNum;
+  }
+};
+
+// Function to check who wins for the round
+var checkWhoWins = function (array) {
+  var resultStatement = "";
+  var maxScoreIndex = 0;
+  var duplicateCount = 0;
+  var currentScore = array[0].playerConcatenateNum;
+
+  // Find the maximum score
+  if (gameVersion == "normal") {
+    for (var counter = 0; counter < array.length; counter += 1) {
+      if (array[counter].playerConcatenateNum > currentScore) {
+        currentScore = array[counter].playerConcatenateNum;
+        maxScoreIndex = counter;
+      }
+    }
+  } else if (gameVersion == "lowestCombined") {
+    for (var counter = 0; counter < array.length; counter += 1) {
+      if (array[counter].playerConcatenateNum < currentScore) {
+        currentScore = array[counter].playerConcatenateNum;
+        maxScoreIndex = counter;
+      }
+    }
+  }
+
+  tempWinner = array[maxScoreIndex];
+
+  resultStatement = `Player ${array[maxScoreIndex].playerNum} won this round!`;
+
+  if (gameMode == "KNOCKOUT" && turnCounter + 1 == playersArray.length) {
+    resultStatement = `Player ${array[maxScoreIndex].playerNum} is the final winner of this round!`;
+  }
+
+  // Find for duplicate scores. It will be a draw if there are duplicate scores
+  for (var counter = 0; counter < array.length; counter += 1) {
+    if (currentScore == array[counter].playerConcatenateNum) {
+      duplicateCount += 1;
+    }
+  }
+  if (duplicateCount >= 2) {
+    resultStatement = `It's a draw! Nobody wins for this round!`;
+    if (gameMode == "KNOCKOUT" && turnCounter + 1 == playersArray.length) {
+      tempWinner = array[0];
+
+      resultStatement = `It's a draw! Player ${array[0].playerNum} had the first player advantage.
+      Thus, player ${array[0].playerNum} is the final winner of this round!`;
+    } else if (gameMode == "KNOCKOUT") {
+      tempWinner = array[0];
+
+      resultStatement = `It's a draw! Player ${array[0].playerNum} had the first player advantage.
+      Thus, player ${array[0].playerNum} won this round!`;
+    }
+  }
+
+  return resultStatement;
+};
+
+// Function to generate leaderboard results
+var generateLeaderBoard = function (array) {
+  var leaderBoardTitle = "<u>Leaderboard</u><br>";
+  var playerResults = "";
+
+  for (var counter = 0; counter < array.length; counter += 1) {
+    playerResults =
+      playerResults +
+      "Player " +
+      array[counter].playerNum +
+      ": " +
+      array[counter].playerSumScore +
+      "<br>";
+  }
+
+  leaderBoardResults = leaderBoardTitle + playerResults;
+  return leaderBoardResults;
+};
+
+// Function to create an object for each player
+
+var createPlayersArray = function (inputPlayersNum) {
+  for (var counter = 0; counter < inputPlayersNum; counter += 1) {
+    var playerObj = {
+      playerNum: counter + 1,
+      playerDiceRollResult: [],
+      playerConcatenateNum: 0,
+      playerSumScore: 0,
+      playerStatus: "NOT PLAYED",
```

Comment:
You could also change playerStatus to playerPlayedStatus so you can use a boolean value instead of a string (usually recommended if theres a true and false for the situation)